marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -359,9 +360,10 @@ handleAction (MarloweEditorAction action) = do
       for_ mContents \contents ->
         sendToSimulation contents
     ME.ViewAsBlockly -> do
+      comesFromBlockly <- use (_marloweEditorState <<< _comesFromBlockly)
```

Comment:
I'm not sure about preserving this as part of the state... can't you figure if it comes from blockly or not via the workflow value?

Reply:
I am afraid not, you can see it is overwritten when pressing the button here: https://github.com/input-output-hk/plutus/blob/34c4781c0d27076390863eab7289438daebe6391/marlowe-playground-client/src/MainFrame/State.purs#L385

And the initialisation of Marlowe Editor doesn't clear the blockly bit, so I have to check, cause otherwise it is possible to undo things like "open example" and "new project".

I don't like this solution either, but is the only way I could think of that doesn't require reworking the whole workflow :/

Reply:
👍🏻